packages/esm-patient-search-app/src/patient-search-overlay/patient-search-overlay.component.tsx

```diff
@@ -14,28 +14,22 @@ interface PatientSearchOverlayProps {
 const PatientSearchOverlay: React.FC<PatientSearchOverlayProps> = ({ onClose, query = '', header }) => {
   const { t } = useTranslation();
   const [searchTerm, setSearchTerm] = useState(query);
-  const handleClear = useCallback(() => setSearchTerm(''), [setSearchTerm]);
-  const showSearchResults = useMemo(() => !!searchTerm?.trim(), [searchTerm]);
+  const showSearchResults = Boolean(searchTerm?.trim());
+  const debouncedSearchTerm = useDebounce(searchTerm);
```

Comment:
A couple of things here:

- We don't need to memoise the call to `searchTerm.trim()`. It's plenty fast. So I've removed that.
- I've replaced the call to lodash's `debounce` function with an implementation based on our custom `useDebounce` function.